app/helpers/searchbar_helper.rb

```diff
@@ -1,14 +1,24 @@
 # frozen_string_literal: true
 module SearchbarHelper
   def self.names_for(educator)
-    authorized_students = Student.with_school.select do |student|
+    authorized_students = educator.school.students.select do |student|
```

Comment:
This is the key line that implements the school-restricting solution @jhilde outlined.